src/sets.jl

```diff
@@ -7,26 +7,30 @@ Abstract supertype for set objects used to encode constraints.
 """
 abstract type AbstractSet end
 
+"""
+    dimension(s::AbstractSet)
+
+Return the underlying dimension (number of vector components) in the set `s`, i.e.,
```

Comment:
"number of vector components" doesn't quite make sense anymore. Please clarify the definition.

Reply:
Hmm, yeah, what should it be?

Typically, mathematical definitions of dimension refer to an intrinsic property of a space (e.g. a vector space). Dimension isn't defined for arbitrary sets. In addition, for `AbstractSet` there are cases like `PositiveSemidefiniteConeTriangle` and `PositiveSemidefiniteConeSquare` where the number we want to return is not the dimension of the vector space to which a given member of the set belongs: same mathematical object (a member of `$S^n_+$`), two different dimensions.

So I guess concrete `AbstractSet` subtypes implicitly define some particular vector space that is used to parameterize members of the set and the `AbstractSet`'s `dimension` is the dimension of that vector space. (Note by the way that the real line is in fact a vector space, and as such it has a well-defined dimension.)

So how about:

> The dimension of the vector space used to parameterize members of `s`.

Reply:
Alternatively, it is equal to the dimension that an `AbstractFunction` should have to be used with the set. Then the dimension of an `AbstractFunction` is 1 for scalar function and the number of rows for a vector function.

Reply:
Whichever you want, just let me know and I'll change it.

Reply:
I think the link with the function is simpler, the vector space might be ambiguous for cases like PSDCTriangle